cmd/flag_completion.go

```diff
@@ -99,13 +98,7 @@ func registerCompletionFuncForFlags(cmd *cobra.Command, o *options) error {
 // that match the toComplete prefix.
 func namespaceCompletionFunc(o *options) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 	return func(_ *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
-		clientConfig := kubernetes.NewClientConfig(o.kubeConfig, o.context)
-		clientset, err := kubernetes.NewClientSet(clientConfig)
-		if err != nil {
-			return compError(err)
-		}
-
-		namespaceList, err := clientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		namespaceList, err := o.client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
```

Comment:
It seems the completion does not work because `o.Complete()` is not called.

```
$ dist/stern __complete --namespace ""
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc8 pc=0x1ff0b5f]

goroutine 1 [running]:
github.com/stern/stern/cmd.registerCompletionFuncForFlags.namespaceCompletionFunc.func1(0x21c1e80?, {0xc00020fd10?, 0xc000401cc0?, 0x2f?}, {0x7ffd900bdeee, 0x0})
        /go/src/github.com/stern/stern/cmd/flag_completion.go:101 +0x3f
github.com/spf13/cobra.(*Command).getCompletions(0xc0003aa900, {0xc000405340, 0xeccd94?, 0x2})
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/completions.go:492 +0xb5a
github.com/spf13/cobra.(*Command).initCompleteCmd.func1(0xc0003aa900, {0xc000405340?, 0x4?, 0x243844d?})
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/completions.go:192 +0x2f
github.com/spf13/cobra.(*Command).execute(0xc0003aa900, {0xc000405340, 0x2, 0x2})
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x863
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003aa600)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(0x0?)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992 +0x13
main.main()
        /go/src/github.com/stern/stern/main.go:32 +0xaf
```

Reply:
I've fixed it.